pkg/kubelet/volumes.go

```diff
@@ -134,6 +142,25 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.Pod) (kubecontainer.VolumeMap,
 		if err != nil {
 			return nil, err
 		}
+		if hasFSGroup && builder.SupportsOwnershipManagement() {
+			filepath.Walk(builder.GetPath(), func(path string, info os.FileInfo, err error) error {
+				if err != nil {
+					return err
+				}
+
+				err = kl.chownRunner.Chown(path, 0, int(fsGroup))
+				if err != nil {
+					glog.Errorf("Chown failed on %v: %v", path, err)
+				}
+
+				err = kl.chmodRunner.Chmod(path, os.FileMode(0770|os.ModeSetgid))
```

Comment:
I think I asked this before but maybe missed the response - is it worthwhile to make a function that sets/clears some bits while preserving others, rather than blindly nuking?  This will mark all files as executable which is unfortunate..